docs/sections/how_to_guides/basic/step/generator_step.md

```diff
@@ -0,0 +1,110 @@
+# GeneratorStep
+
+The [`GeneratorStep`][distilabel.steps.GeneratorStep] is a subclass of [`Step`][distilabel.steps.Step] that is intended to be used as the first step within a [`Pipeline`][distilabel.pipeline.Pipeline], because it doesn't require input and generates data that can be used by other steps. Alternatively, it can also be used as a standalone.
+
+```python
+from typing import List
+from typing_extensions import override
+
+from distilabel.steps import GeneratorStep
+from distilabel.steps.typing import GeneratorStepOutput
+
+class MyGeneratorStep(GeneratorStep):
+    instructions: List[str]
+
+    @override
+    def process(self, offset: int = 0) -> GeneratorStepOutput:
+        if offset:
+            self.instructions = self.instructions[offset:]
+
+        while self.instructions:
+            batch = [
+                {
+                    "instruction": instruction
+                } for instruction in self.instructions[: self.batch_size]
+            ]
+            self.instructions = self.instructions[self.batch_size :]
+            yield (
+                batch,
+                True if len(self.instructions) == 0 else False,
+            )
+
+    @property
+    def outputs(self) -> List[str]:
+        return ["instruction"]
+```
+
+Then we can use it as follows:
+
+```python
+step = MyGeneratorStep(
+    name="my-generator-step",
+    instructions=["Tell me a joke.", "Tell me a story."],
+    batch_size=1,
+)
+step.load()
+
+next(step.process(offset=0))
+# ([{'instruction': 'Tell me a joke.'}], False)
+next(step.process(offset=1))
+# ([{'instruction': 'Tell me a story.'}], True)
+```
+
+!!! NOTE
+    The `Step.load()` always needs to be executed when being used as a standalone. Within a pipeline, this will be done automatically during pipeline execution.
+
+## Defining custom GeneratorSteps
+
+We can define a custom generator step by creating a new subclass of the [`GeneratorStep`][distilabel.steps.GeneratorStep] and defining the following:
+
+- `outputs`: is a property that returns a list of strings with the names of the output fields.
+
+- `process`: is a method that yields output data and a boolean flag indicating whether that's the last batch to be generated.
+
+!!! NOTE
+    The default signature for the `process` method is `process(self, *inputs: StepInput) -> StepOutput`. The argument `inputs` should be respected, no more arguments can be provided, and the type-hints and return type-hints should be respected too because it should be able to receive any number of inputs by default i.e. more than one [`Step`][distilabel.steps.Step] at a time could be connected to the current one.
```

Comment:
The old text that we had here it's important. For the `GeneratorStep`s steps the signature of the process method must be `process(self, offset: int = 0) -> GeneratorStepOutput`.

Reply:
done